go/vt/vtgate/planbuilder/limit.go

```diff
@@ -127,6 +123,18 @@ func (l *limit) SetLimit(limit *sqlparser.Limit) error {
 	}
 	l.elimit.Count = pv
 	l.input.SetUpperLimit(count)
+
+	offset, ok := limit.Offset.(*sqlparser.SQLVal)
+	if ok {
```

Comment:
Better to move this at the beginning and make the non-offset part as the else (or make this block return early).
Otherwise, we'll be calling `SetUpperLimit` twice.